### PR TITLE
Firebase structure

### PR DESCRIPTION
**Pull request by pablobaldez**
#### The problem

My friend Henrique Delgado, also known as Henriqueta bum bum guloso, has committed in the main branch without opening a PR for that.
Now I will open some PRs to change some points of implementation, adding comments to explain why/what I'm changing. One of the things we should fix is the firebase collections and how we can organize the info there

#### The solution

In this PR you can see I'm changing the structure of collections on firebase and also I'm using a batched write to do it.
See the comments in the code for more details

#### Next steps
Remove the firebase dependency from ViewModel and move it to a repository-usecase style

### CODE REVIEW

File: app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt [hunk 1]
```diff
@@ -50,8 +50,99 @@ class CreateGroupViewModel(
...
+            mapOf(
+                // cuidado com esse !!, mas por hora vamos fazer pra simplificar
+                "ref" to firestore.collection("users").document(userAuth.getId()!!),
+                "name" to userAuth.get()!!.name,
+                "image" to userAuth.get()!!.imageUrl
+            )
+        )
+
+        // agora nós criamos o documento dentro da coleção do usuário logado
+        // assim saberemos todos os grupos que o usuário é membro.
```
Comment (HenriqueDelgad0) [inline app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt:128]: Pode crer pode crer

File: app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt [hunk 2]
```diff
@@ -50,8 +50,99 @@ class CreateGroupViewModel(
...
+        // a tela que lista os grupos deve buscar os grupos dessa coleção
+        val membershipRef = firestore.document(userAuth.getId()!!)
```
Comment (HenriqueDelgad0) [inline app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt:130]: Esse carinha aqui ta quebrando no app, acho que faltou a collection de users antes de document. 
Tipo ->  firestore.collection("users").document(userAuth.getId()!!)

File: app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt [hunk 3]
```diff
@@ -50,8 +50,99 @@ class CreateGroupViewModel(
...
+            .collection("membership")
+            .document()
+        batch.set(
+            membershipRef,
+            mapOf(
+                // repete as infos básicas do grupo
+                "name" to groupName,
+                "url" to "www.porntube.com/kidbengala.jpg",
```
Comment (HenriqueDelgad0) [inline app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt:138]: ashashuauhsausuhs vtf

**Review by HenriqueDelgad0 — APPROVED** [review]

File: app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt (hunk 2 above)
Comment (pablobaldez) [reply app/src/main/java/br/com/mob1st/bet/features/groups/presentation/createGroup/CreateGroupViewModel.kt:130]: sim!!!! nice catch!!!
o outro PR tem alterações nessa parte, entao tu nao precisa corrigir isso neste PR aqui

**Pull request merged**